src/sage/combinat/symmetric_group_algebra.py

```diff
@@ -1885,7 +1885,7 @@ def seminormal_basis(self, mult='l2r'):
                     basis.append(self.epsilon_ik(t1, t2, mult=mult))
         return basis
 
-    def dft(self, form="seminormal", mult='l2r'):
```

Comment:
According to Sage policy, there has to be a one year deprecation warning period before changing APIs. So the `form` argument cannot be removed yet, and instead a [deprecation](https://doc.sagemath.org/html/en/reference/misc/sage/misc/superseded.html#sage.misc.superseded.deprecation) should be thrown. (Source: [Developer Guide](https://doc.sagemath.org/html/en/developer/coding_in_python.html#deprecation))

Reply:
I've added a deprecation warning to the "seminormal" form which states that it does not work when p|n!, but that "form=modular" does. Ideally the different forms would agree when p \nmid n!, but I haven't thought about that yet.